### ISSUE TITLE
Do not restart after updates

### ISSUE THREAD

**Issue by zenmaster24** [rgl/packer-plugin-windows-update#127]
Hi,

Having an issue where the recent windows update KB5025230 is causing the instance to take a long time to reboot - Packer errors with the message `retry count exhausted. Last err: http response error: 401 - invalid content type`.

Is it possible to set the windows-update provisioner to *not* reboot after the updates are applied, and instead handle a restart with the windows-restart provisioner which has a restart_timeout parameter?

Thanks

**Comment by rgl** [comment]
No. Handling restarts is a typical requirement for applying windows updates. It has to be handled by this provisioner because packer does not have loops.

Did you try the [restart_timeout parameter on this provisioner](https://github.com/rgl/packer-plugin-windows-update/blob/6793f6375aa3b786be6ef36d571dc9d35f9f90de/update/provisioner.go#L47))?

**Comment by zenmaster24** [comment]
I did - didnt work. You can use the flag `IgnoreReboot` with Install-WindowsUpdate - are you saying this provisioner will persist across reboots requested by an update that has been installed?

**Comment by rgl** [comment]
> > Did you try the [restart_timeout parameter on this provisioner](https://github.com/rgl/packer-plugin-windows-update/blob/6793f6375aa3b786be6ef36d571dc9d35f9f90de/update/provisioner.go#L47))?
> 
> I did - didnt work.

Oh that's a bummer :-(

> > It has to be handled by this provisioner because packer does not have loops.
> 
> You can use the flag `IgnoreReboot` with Install-WindowsUpdate - are you saying this provisioner will persist across reboots requested by an update that has been installed?

Not sure what you mean be `persist`, but, this provider will loop installing updates and rebooting until the updates are installed.

**Comment by JoeBrunsTR** [comment]
The problem with insisting on restarting on every single update that is done is wasting resources.  Simply put, why boot 3 times when I can only do it once.

  provisioner "windows-update" {
    search_criteria = "IsInstalled=0"
    filters         = ["include:$_.Title -like '*Servicing Stack Update*'"]
    update_limit    = 25
    update_reboot   = false
  }

  provisioner "windows-update" {
    search_criteria = "IsInstalled=0"
    filters         = ["include:$_.Title -like '*Cumulative Update*'"]
    update_limit    = 25
    update_reboot   = false
  }

  provisioner "windows-update" {
    search_criteria = "IsInstalled=0"
    filters         = ["include:$_.Title -like '*Windows Malicious Software Removal Tool*'"]
    update_limit    = 25
    update_reboot   = false
  }

  provisioner "windows-restart" {
    pause_before          = "10s"
    restart_check_command = "powershell -command \"& {Write-Output 'SSU and Cumulative Update reboot'}\""
  }

**Comment by Flip0188** [comment]
> The problem with insisting on restarting on every single update that is done is wasting resources. Simply put, why boot 3 times when I can only do it once.
> 
> provisioner "windows-update" { search_criteria = "IsInstalled=0" filters = ["include:$_.Title -like '_Servicing Stack Update_'"] update_limit = 25 update_reboot = false }
> 
> provisioner "windows-update" { search_criteria = "IsInstalled=0" filters = ["include:$_.Title -like '_Cumulative Update_'"] update_limit = 25 update_reboot = false }
> 
> provisioner "windows-update" { search_criteria = "IsInstalled=0" filters = ["include:$_.Title -like '_Windows Malicious Software Removal Tool_'"] update_limit = 25 update_reboot = false }
> 
> provisioner "windows-restart" { pause_before = "10s" restart_check_command = "powershell -command "& {Write-Output 'SSU and Cumulative Update reboot'}"" }

What resources are being wasted, and why invoke the provisioner 3 times instead of a single entry with multiple filters and let Windows and the provisioner handle the reboots as needed?

provisioner "windows-update" {
    pause_before    = "60s"
    search_criteria = "IsInstalled=0"
    filters = [
      "include:$_.Title -like 'Servicing Stack Update'",
      "include:$_.Title -like 'Cumulative Update'",
      "include:$_.Title -like 'Windows Malicious Software Removal Tool'",
      "exclude:$_.Title -like '*Preview*'",
      "exclude:$_.InstallationBehavior.CanRequestUserInput"
    ]
    restart_timeout = "120m"
  }
